src/components/card/AttachmentList.vue

```diff
@@ -0,0 +1,223 @@
+<!--
+  - @copyright Copyright (c) 2020 Jakob Röhrl <jakob.roehrl@web.de>
+  -
+  - @author Jakob Röhrl <jakob.roehrl@web.de>
+  -
+  - @license GNU AGPL version 3 or any later version
+  -
+  - This program is free software: you can redistribute it and/or modify
+  - it under the terms of the GNU Affero General Public License as
+  - published by the Free Software Foundation, either version 3 of the
+  - License, or (at your option) any later version.
+  -
+  - This program is distributed in the hope that it will be useful,
+  - but WITHOUT ANY WARRANTY; without even the implied warranty of
+  - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  - GNU Affero General Public License for more details.
+  -
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program. If not, see <http://www.gnu.org/licenses/>.
+  -
+  -->
+
+<template>
+	<div class="attachment-list">
+		<ul>
+			<li v-for="attachment in attachments"
+				:key="attachment.id"
+				class="attachment">
+				<a class="fileicon" :style="mimetypeForAttachment(attachment.extendedData.mimetype)" :href="attachmentUrl(attachment)" />
+				<div class="details">
+					<a :href="attachmentUrl(attachment)" target="_blank">
+						<div class="filename">
+							<span class="basename">{{ attachment.data }}</span>
+						</div>
+						<span class="filesize">{{ formattedFileSize(attachment.extendedData.filesize) }}</span>
+						<span class="filedate">{{ relativeDate(attachment.createdAt*1000) }}</span>
+						<span class="filedate">{{ attachment.createdBy }}</span>
+					</a>
+				</div>
+				<Actions v-if="selectable">
+					<ActionButton icon="icon-confirm" @click="addAttachment(attachment)">
+						{{ t('deck', 'Add this attachment') }}
+					</ActionButton>
+				</Actions>
+				<Actions v-if="removable">
+					<ActionButton v-if="attachment.deletedAt === 0" icon="icon-delete" @click="deleteAttachment(attachment)">
+						{{ t('deck', 'Delete Attachment') }}
+					</ActionButton>
+
+					<ActionButton v-else icon="icon-history" @click="restoreAttachment(attachment)">
+						{{ t('deck', 'Restore Attachment') }}
+					</ActionButton>
+				</Actions>
+			</li>
+		</ul>
+	</div>
+</template>
+
+<script>
+import { Actions, ActionButton } from '@nextcloud/vue'
+import relativeDate from '../../mixins/relativeDate'
+import { formatFileSize } from '@nextcloud/files'
+
+export default {
+	name: 'AttachmentList',
+	components: {
+		Actions,
+		ActionButton,
+	},
+	mixins: [relativeDate],
+	props: {
+		cardId: {
+			type: Number,
+			required: true,
+		},
+		editor: {
```

Comment:
Hm, I'm not a big fan of that appraoch as this bundles the AttachmentList with the implementation of the editor. Having an event would be much nicer here